common/VotingUtils.js

```diff
@@ -17,8 +24,15 @@ const constructCommitment = async (request, roundId, web3, price, account) => {
   const hash = web3.utils.soliditySha3(priceWei, salt);
 
   const vote = { price: priceWei, salt };
-  const { publicKey } = await deriveKeyPairFromSignatureTruffle(web3, getKeyGenMessage(roundId), account);
-  const encryptedVote = await encryptMessage(publicKey, JSON.stringify(vote));
+  let derivedPublicKey;
+  if (argv.network === "metamask") {
+    const { publicKey } = await deriveKeyPairFromSignatureMetamask(web3, getKeyGenMessage(roundId), account);
```

Comment:
If you change `derivedPublicKey` back to `publicKey`, you can do this assignment like this:
```
({ publicKey } = await deriveKeyPairFromSignatureMetamask(web3, getKeyGenMessage(roundId), account));
```
or like this:
```
publicKey = (await deriveKeyPairFromSignatureMetamask(web3, getKeyGenMessage(roundId), account)).publicKey;
```

Reply:
nice saves a variable

Reply:
done